Goesdump enhancement (Fix #11 opensatelliteproject/goesdump)

n2cr opened the pull request:
Adds 2 new keywords to the config file:

"SysLogServer" - The host name or IP address or the remote syslog server.
                            Default is "localhost".

"SysLogFacility" - The syslog facility (see <sys/syslog.h>) which should be
                             used in syslog messages.  Default is "LOG_USER".

n2cr committed:
Goesdump enhancement (Fix #11 opensatelliteproject/goesdump) which adds 2 new
keywords to the config file:

"SysLogServer" - The host name or IP address or the remote syslog server.
                 Default is "localhost".

"SysLogFacility" - The syslog facility (see <sys/syslog.h>) which should be
                   used in syslog messages.  Default is "LOG_USER".

racerxdl commented on XRIT/Log/SyslogClient.cs:
```diff
@@ -40,12 +41,36 @@ public class SyslogClient {
             }
         }
+
+        public static void InitFacilityMap() {
```
That should probably be private to avoid issues if someone calls it somwhere.

n2cr replied:
Agreed.  Will fix.

racerxdl commented on goesdump/HeadlessMain.cs:
```diff
@@ -36,6 +36,7 @@ public class HeadlessMain {
 
         private static List<ConsoleMessage> messageList = new List<ConsoleMessage>();
         private static Mutex messageListMutex = new Mutex();
+        private SyslogClient syslog = new SyslogClient();
```
You don't need to create it here, since the instance on headless main is just for testing if the syslog is configured.

n2cr committed:
Goesdump enhancement (Fix #11 opensatelliteproject/goesdump)

Address comments from code reviewer.  Eliminate dynamic allocation of
SyslogClient where it it not necessary.  Make InitFacilityMap() private.

racerxdl commented:
That will also close https://github.com/opensatelliteproject/OpenSatelliteProject/issues/9
